### PR TITLE
refactor(form-builder): use context for conditional read only

### PR DESCRIPTION
**Pull request by sjelfull**
### Description

This PR changes the `ConditionalReadOnlyField` field to use a context to pass the evaluated `readOnly` value instead of mapping over the children with `React.Children.map`. I wasn't able to prove that mapping over the children caused a performance regression, but it's still a good change to make.

### What to review

- That tests still run successfully

### Notes for release

n/a

### CODE REVIEW

**Review by vicbergquist — CHANGES_REQUESTED** [review]
Great job! Looks good to me. This should also probably be added to the change resolver?

**Comment by sjelfull** [comment]
> Great job! Looks good to me. This should also probably be added to the change resolver?

Ah yeah, missed that! Will sort out.

File: packages/@sanity/form-builder/src/inputs/common/conditionalReadOnly/ConditionalReadOnlyContextProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React, {useMemo} from 'react'
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/inputs/common/conditionalReadOnly/ConditionalReadOnlyContextProvider.tsx:1]: How does this relate to the `ConditionalReadOnlyContextProvider` in @sanity/base? Do we still need both?

File: packages/@sanity/form-builder/src/inputs/common/conditionalReadOnly/ConditionalReadOnlyContext.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import {createContext} from 'react'
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/inputs/common/conditionalReadOnly/ConditionalReadOnlyContext.ts:1]: How does this relate to the ConditionalReadOnlyContext exported from @sanity/base? Do we still need both?

**Review by bjoerge — COMMENTED** [review]
I'm wondering if we still need the ConditionalReadOnly context files in the form-builder package, but other than that LGTM!

File: packages/@sanity/form-builder/src/inputs/common/conditionalReadOnly/ConditionalReadOnlyContextProvider.tsx (hunk 1 above)
Comment (sjelfull) [reply packages/@sanity/form-builder/src/inputs/common/conditionalReadOnly/ConditionalReadOnlyContextProvider.tsx:1]: I believe we don't need this any more, will remove them

**Review by RitaDias — APPROVED** [review]
🛳 do the thing

**Pull request merged**